### PR TITLE
Deprecate the include_in_all mapping parameter

### PR DESCRIPTION
**Pull request by dakrone**
Relates to #22970

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java [hunk 1]
```diff
@@ -254,6 +256,8 @@ public static void parseField(FieldMapper.Builder builder, String name, Map<Stri
                     throw new MapperParsingException("include_in_all in multi fields is not allowed. Found the include_in_all in field ["
                         + name + "] which is within a multi field.");
                 } else {
+                    deprecationLogger.deprecated("field [include_in_all] is deprecated, as [_all] is deprecated, " +
+                                    "and will be disallowed in the next major release, use [copy_to] instead.");
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java]: does it make sense to say 6.0 explicitly to avoid ambiguities?

**Review by jpountz — APPROVED** [review]
LGTM

File: core/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java (hunk 1 above)
Comment (dakrone) [inline core/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java]: Sure, I can change it to be 6.0

**Comment by dakrone** [comment]
Thanks @jpountz!

**Pull request merged**